src/vivarium/framework/values.py

```diff
@@ -354,10 +432,10 @@ def _register_value_producer(
     def register_value_modifier(
         self,
         value_name: str,
-        modifier: Callable,
-        requires_columns: List[str] = (),
-        requires_values: List[str] = (),
-        requires_streams: List[str] = (),
+        modifier: ValueSource | ValueMutator,
```

Comment:
This can really also be a `ValueSource`?

Reply:
From the perspective of a protocol, I would say that that is the case:

> If the pipeline has a ``replace_combiner``, the
>             modifier should accept the same arguments as the pipeline source
>             with an additional last positional argument for the results of the
>             previous stage in the pipeline. For the ``list_combiner`` strategy,
>             the pipeline modifiers should have the same signature as the pipeline
>             source.

Though arguably the protocols aren't quite aptly named in this case.